### PR TITLE
Bug fix - `azurerm_express_route_connection` - Remove `private_link_fast_path_enabled `default vaule

### LINKED ISSUE #26746: `azurerm_express_route_connection` updating Express Route Connection fails with `400 Bad Request`
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.6.3

### AzureRM Provider Version

3.113.0

### Affected Resource(s)/Data Source(s)

azurerm_express_route_connection

### Terraform Configuration Files

```hcl
resource "azurerm_express_route_connection" "main" {
  for_each                         = local.express_route_connections
  name                             = each.value.name
  express_route_gateway_id         = azurerm_express_route_gateway.main[each.value.hub_key].id
  express_route_circuit_peering_id = each.value.peering_id
  routing_weight                   = each.value.weight
  enable_internet_security = false

  routing {
    associated_route_table_id = azurerm_virtual_hub_route_table.main[each.value.hub_key].id
  }
}
```

### Debug Output/Panic Output

```shell
│ Error: updating Express Route Connection (Subscription: "xxx"
│ Resource Group Name: "xxx"
│ Express Route Gateway Name: "xxx"
│ Express Route Connection Name: "xxx"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: InvalidRequestFormat: Cannot parse the request.
│ 
│   with azurerm_express_route_connection.main["xxx"],
│   on main.tf line 305, in resource "azurerm_express_route_connection" "main":
│  305: resource "azurerm_express_route_connection" "main" {
```

### Expected Behaviour

_No response_

### Actual Behaviour

_No response_

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by xuzhang3**
The `private_link_fast_path_enabled`  should not be added to this resource. Don't know why the API have this property in the API doc. `private_link_fast_path_enabled` is available the Virtual Network Gateway(**ExpressRoute**)  but this resource(`azurerm_express_route_connection`) is for **ExpressRouteGateway**, they are different resources. 

The Virtual Network Gateway(ExpressRoute) can be created through the Virtual Network Gateway,  (Gateway type: `ExpressRoute`). The `ExpressRouteGateway` is  created though the `Virtual Wan -> Hubs`, you can find the ExpressRoute Gateway after the hub is created.
image

## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [ ] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_resource` - support for the `thing1` property [GH-00000]

This is a (please select all that apply):

- [x] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #26746

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Comment by xuzhang3** [comment]
```log
------- Stdout: -------
=== RUN   TestAccExpressRouteConnection
=== RUN   TestAccExpressRouteConnection/Resource
=== RUN   TestAccExpressRouteConnection/Resource/basic
=== RUN   TestAccExpressRouteConnection/Resource/requiresImport
=== RUN   TestAccExpressRouteConnection/Resource/complete
=== RUN   TestAccExpressRouteConnection/Resource/update
    testcase.go:128: Step 3/4 error: Error running apply: exit status 1
        Error: creating Route Map (Subscription: "*******"
        Resource Group Name: "acctestRG-erconnection-240806121225941266"
        Virtual Hub Name: "acctest-vhub-240806121225941266"
        Route Map Name: "routemapfirst"): polling after RouteMapsCreateOrUpdate: context deadline exceeded
          with azurerm_route_map.routemap1,
          on terraform_plugin_test.tf line 93, in resource "azurerm_route_map" "routemap1":
          93: resource "azurerm_route_map" "routemap1" {
        creating Route Map (Subscription: "*******"
        Resource Group Name: "acctestRG-erconnection-240806121225941266"
        Virtual Hub Name: "acctest-vhub-240806121225941266"
        Route Map Name: "routemapfirst"): polling after RouteMapsCreateOrUpdate:
        context deadline exceeded
--- FAIL: TestAccExpressRouteConnection (23037.43s)
    --- FAIL: TestAccExpressRouteConnection/Resource (23037.43s)
        --- PASS: TestAccExpressRouteConnection/Resource/basic (5392.82s)
        --- PASS: TestAccExpressRouteConnection/Resource/requiresImport (5275.42s)
        --- PASS: TestAccExpressRouteConnection/Resource/complete (5279.29s)
        --- FAIL: TestAccExpressRouteConnection/Resource/update (7089.90s)
FAIL
```

**Comment by xuzhang3** [comment]
```log
=== RUN   TestAccExpressRouteConnection/Resource
=== RUN   TestAccExpressRouteConnection/Resource/update
--- PASS: TestAccExpressRouteConnection (7852.24s)
    --- PASS: TestAccExpressRouteConnection/Resource (7852.24s)
        --- PASS: TestAccExpressRouteConnection/Resource/update (7852.24s)
```

**Comment by xuzhang3** [comment]
All test passed, but some of the tests are relate to `azurerm_route_map` which may result in timeout. Increase the timeout of `azurerm_route_map`  from 30 mins to 60 mins

File: internal/services/network/express_route_connection_resource.go [hunk 1]
```diff
@@ -197,10 +192,17 @@ func resourceExpressRouteConnectionCreate(d *pluginsdk.ResourceData, meta interf
 			RoutingConfiguration:      expandExpressRouteConnectionRouting(d.Get("routing").([]interface{})),
 			RoutingWeight:             pointer.To(int64(d.Get("routing_weight").(int))),
 			ExpressRouteGatewayBypass: pointer.To(d.Get("express_route_gateway_bypass_enabled").(bool)),
-			EnablePrivateLinkFastPath: pointer.To(d.Get("private_link_fast_path_enabled").(bool)),
 		},
 	}
 
+	privateLinkFatPath := d.GetRawConfig().AsValueMap()["private_link_fast_path_enabled"]
```
Comment (stephybun) [inline internal/services/network/express_route_connection_resource.go:198]: Could we be mindful of the spelling here 
```suggestion
	privateLinkFastPath := d.GetRawConfig().AsValueMap()["private_link_fast_path_enabled"]
```

File: internal/services/network/express_route_connection_resource.go [hunk 2]
```diff
@@ -293,10 +292,17 @@ func resourceExpressRouteConnectionUpdate(d *pluginsdk.ResourceData, meta interf
...
```
Comment (stephybun) [inline internal/services/network/express_route_connection_resource.go:298]:
```suggestion
	privateLinkFastPath := d.GetRawConfig().AsValueMap()["private_link_fast_path_enabled"]
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @xuzhang3. Could you fix the spelling of the variable please?

File: internal/services/network/express_route_connection_resource.go (hunk 1 above)
Comment (xuzhang3) [reply internal/services/network/express_route_connection_resource.go]: fixed

**Review by stephybun — APPROVED** [review]
Thanks @xuzhang3 LGTM 👍

**Pull request merged**